site/content/blog/carvelizing-helm-chart.md

```diff
@@ -0,0 +1,296 @@
+---
+title: "Carvelizing Helm Chart"
+slug: carvelize-helm-chart
+date: 2022-02-16
+author: Rohit Aggarwal
+excerpt: "Use Carvel to author and consume helm chart "
+image: /img/logo.svg
+tags: ['carvel', 'helm', 'gitops']
+---
+
+In this blog, we will see how to author [`bitnami nginx helm chart`](https://github.com/bitnami/charts/tree/master/bitnami/nginx) into Carvel package and then consume the same.
+
+## Why should I choose Carvel
+
+Kubernetes configuration takes many forms – plain YAML configurations, Helm charts, ytt templates, jsonnet templates, etc. Software running on Kubernetes lives in many different places: a Git repository, an archive over HTTP, a Helm repository, etc.
+
+Kapp-controller, a carvel tool, provide software authors flexibility to choose their own configuration tools, while providing software consumers with consistent declarative APIs to customize, install, and update software on Kubernetes from various sources.
+
+## Pre Requisite
+
+Basic knowledge of imgpkg, kbld, kapp controller
+
+[`imgpkg`](https://carvel.dev/imgpkg/): A tool to package, distribute, and relocate your Kubernetes configuration and dependent OCI images as one OCI artifact: a bundle.
+
+[`kbld`](https://carvel.dev/kbld/): kbld incorporates image building and image pushing into your development and deployment workflows.
+
+[`kapp-controller`](https://carvel.dev/kapp-controller/): Kapp-controller provides declarative APIs to create, customize, install, and update your Kubernetes applications into packages.
+
+## Installing Carvel Tools
+
+We'll be using Carvel tools throughout this tutorial, so first we'll install them:
+
+Install the tools with the scripts below:
+To install all the carvel tools, run this `install.sh` script.
+
+```bash
+$ wget https://raw.githubusercontent.com/vmware-tanzu/carvel-kapp-controller/83fffcfe99a65031b4170813acf94f8d5058b346/hack/dependencies.yml
+$ wget https://raw.githubusercontent.com/vmware-tanzu/carvel-kapp-controller/83fffcfe99a65031b4170813acf94f8d5058b346/hack/install-deps.sh
+$ chmod a+x ./install-deps.sh
+$ ./install-deps.sh
+$ kubectl apply -f https://github.com/vmware-tanzu/carvel-kapp-controller/releases/latest/download/release.yml
+```
+
+**Note**: Going forward, I will be using [`kapp`](https://carvel.dev/kapp/) to deploy the yaml. `kapp` is a cli used to deploy and view groups of Kubernetes resources as “application”.
```

Comment:
should this be then a prereq?

Reply:
Adopted. As I am not giving equivalent `kubectl` commands, I think its right to make it as pre-requisite.